src/app/app-interactive-map/app-interactive-map.component.ts

```diff
@@ -127,23 +134,90 @@ export class AppInteractiveMapComponent implements OnInit, AfterViewInit {
+    this.store.dispatch(new fromActions.ReactionOperation({
+      item: reactionId,
+      operationTarget: 'knockoutReactions',
+      direction: this.card.knockoutReactions.includes(reactionId) ? OperationDirection.Undo : OperationDirection.Do,
+    }));
+  }
+
+  handleSetAsObjective(reactionId: string): void {
+    if (this.card.objectiveReaction && this.card.objectiveReaction.reactionId === reactionId) {
+      this.store.dispatch(new fromActions.SetObjectiveReaction({
+        reactionId: null,
+        direction: null,
+      }));
+    } else {
+      this.store.dispatch(new fromActions.SetObjectiveReaction({
+        reactionId,
+        direction: 'max',
+      }));
+    }
+  }
+
+  handleChangeBounds(reactionId: string): void {
+    this.store.dispatch(new fromActions.ReactionOperation({
+      item: {
+        reaction: this.card.model.reactions.find((r) => r.id === reactionId),
+        lowerBound: parseInt(this.elRef.nativeElement.querySelector('#lowerbound').value, 10),
+        upperBound: parseInt(this.elRef.nativeElement.querySelector('#upperbound').value, 10),
+      },
+      operationTarget: 'bounds',
+      direction: OperationDirection.Do,
+    }));
+  }
+
+  handleResetBounds(reactionId: string): void {
+    this.store.dispatch(new fromActions.ReactionOperation({
+      item: <Bound>(this.findBoundReaction(reactionId)),
+      operationTarget: 'bounds',
+      direction: OperationDirection.Undo,
+    }));
+  }
+
+  handleObjectiveDirection(reactionId: string): void {
+    this.store.dispatch(new fromActions.SetObjectiveReaction({
+      reactionId,
+      direction: this.card.objectiveReaction.direction === 'max' ? 'min' : 'max',
+    }));
+  }
+
+  checkStateObjective(reactionId: string): boolean {
+    if (this.card.objectiveReaction) {
+      return (this.card.objectiveReaction.reactionId === reactionId && this.card.objectiveReaction.direction === 'max');
+    } else {
+      return false;
+    }
+  }
+
+  drawSwitch (reactionId: string): string {
+    if (this.card.objectiveReaction) {
+      return this.card.objectiveReaction.reactionId === reactionId ? '' : 'hidden';
+    } else {
+      return 'hidden';
+    }
+  }
+
+  upperBound(reactionId: string): number {
+    const reaction = this.findBoundReaction(reactionId);
+    return reaction ? reaction.upperBound : this.card.model.reactions.find((r) => r.id === reactionId).upper_bound;
   }
 
-  handleSetAsObjective(args: string): void {
-    console.log('SET AS OBJECTIVE', args);
+  lowerBound(reactionId: string): number {
+    const reaction = this.findBoundReaction(reactionId);
+    return reaction ? reaction.lowerBound : this.card.model.reactions.find((r) => r.id === reactionId).lower_bound;
   }
 
-  handleChangeBounds(args: string): void {
-    console.log('CHANGE BOUNDS', args);
+  knockoutText(reactionId: string): string {
+    return this.card.knockoutReactions && this.card.knockoutReactions.includes(reactionId) ? 'Undo knockout' : 'Knockout';
```

Comment:
I am a bit conserned with defining the actual button text in Caffeine - doesn't that logic belong in the CustomTooltip component in our escher fork? While Caffeine should only provide the information of whether the reaction currently is knocked out or not.

Same for the objective below.